includes/class-wc-stripe-api.php

```diff
@@ -265,6 +286,20 @@ public static function request( $request, $api = 'charges', $method = 'POST', $w
 	 * @param string $api
 	 */
 	public static function retrieve( $api ) {
+		// If keep count of consecutive 401 errors, and it exceeds INVALID_API_KEYS_COUNT_THRESHOLD,
+		// we return null until the cache expires (INVALID_API_KEYS_DELAY_IN_SECONDS) or the keys are updated.
+		$invalid_api_keys_count = WC_Stripe_Database_Cache::get( self::INVALID_API_KEYS_CACHE_KEY );
+		if ( ! empty( $invalid_api_keys_count ) && self::INVALID_API_KEYS_COUNT_THRESHOLD <= $invalid_api_keys_count ) {
+			// We skip logging the error here because when there is no Account cache,
+			// the instantiation of the UPE gateway triggers a call to this method for
+			// every available payment method. This would result in excessive log entries
+			// which is not useful.
+			// We only log the error when the count exceeds the threshold.
```

Comment:
Nit: it feels like we could be more accurate with this comment.
```suggestion
			// We only log the error when the count exceeds the threshold for the first time.
```